pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -146,5 +148,20 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	if err := validateKubeletOSConfiguration(kc); err != nil {
 		allErrors = append(allErrors, err)
 	}
+
+	if kc.ReservedSystemCPUs != "" {
+		// --reserved-cpus does not support --system-reserved-cgroup or --kube-reserved-cgroup
+		if kc.SystemReservedCgroup != "" || kc.KubeReservedCgroup != "" {
+			allErrors = append(allErrors, fmt.Errorf("can't use --reserved-cpus with --system-reserved-cgroup or --kube-reserved-cgroup"))
+		}
+		if matched, _ := regexp.MatchString(`[^-,0-9]`, kc.ReservedSystemCPUs); matched {
+			// if ReservedSystemCPUs contains any character other than [-,0-9]
+			allErrors = append(allErrors, fmt.Errorf("invalid configuration: --reserved-cpus must be in a string of digits separated by comma or -, not %q", kc.ReservedSystemCPUs))
+		}
+		if _, err := cpuset.Parse(kc.ReservedSystemCPUs); err != nil {
```

Comment:
does this parse not catch the invalid characters checked for above?

Reply:
consolidated the validation into one place. pls check again.

Reply:
if you're only going to do the validation in one place, checking for well-formed options is better to do in validation.go, so that we can validate a config without actually starting a kubelet

Reply:
updated as we discussed.